### ISSUE TITLE
Bug: profile not recognized

### ISSUE THREAD

**Issue by ggabernet** [seqeralabs/nf-tower#300]
Hi,

when using the Tower Launcher, a profile that is imported into `nextflow.config` is not recognized by the Tower launcher. In case this is relevant, the profile is defined in the `dsl2` branch but not yet merged into the master branch.

https://github.com/nf-core/bcellmagic/blob/fba2622fefca465e351ed5e85f133863fca1303f/nextflow.config#L172

The profile does not appear in the drop-down menu:
![Screenshot 2021-05-04 at 20 36 03](https://user-images.githubusercontent.com/3973503/117053697-8e698600-ad19-11eb-867e-eb40176f81a3.png)

And if trying to force it anyways, it fails to launch with this message:
![Screenshot 2021-05-04 at 20 45 16](https://user-images.githubusercontent.com/3973503/117053840-bfe25180-ad19-11eb-95a6-621f6bfe954a.png)

**Comment by pditommaso** [comment]
I Gisella, I've tried with the new beta ([here](https://staging-tower.xyz/)) and it looks the test_full is accepted, tho still not listed in the dropdown. @jordeu you may want to give a try, my understanding is that with the config parsed it should be detected .

**Comment by jordeu** [comment]
We have an error here, but it's not related with the parser it self. The problem is that only the profiles at master are used to do this checking, so given that `test_full` it's not at master and only at `dsl2` branch this is why it complains with the `unknown pipeline profile`. We are going to fix this asap, but meanwhile if you want a workaround one option would be to create an empty `test_full` profile at master branch.

**Comment by ggabernet** [comment]
Ok thanks for checking it out!

**Comment by drpatelh** [comment]
Yep, I am seeing a similar thing and was a little confused too but it makes sense now if only the `profile` definitions on `master` are pulled in.

In my case, I was trying to use [these two `profile` definitions](https://github.com/nf-core/viralrecon/blob/3078442993edfd0dee59239ea2572ea6e77d4e2e/nextflow.config#L202-L203) that are only on the `dev` branch and not on `master` and so was unable to find them in the profile listing in Tower:

![image](https://user-images.githubusercontent.com/23529759/118107391-99d24680-b3d6-11eb-89e4-bf4c03cbc6b8.png)

**Comment by pditommaso** [comment]
ok, currently the profile names are only taken from the master branch. It needs to be solved.
